src/ligature.rs

```diff
@@ -0,0 +1,146 @@
+use skrifa::{
+    charmap::Charmap,
+    raw::{
+        tables::gsub::{CoverageTable, LigatureMarker, LigatureSet, SubstitutionSubtables},
+        types::Offset16,
+        ArrayOfOffsets, FontRef, ReadError, TableProvider, TableRef,
+    },
+    GlyphId,
+};
+use std::{cell::RefCell, iter};
+
+use crate::error::IconResolutionError;
+
+pub struct Ligature {
+    // Includes all components, not just 1..N.
```

Comment:
This is a non-standard definition of components and one that will only make sense to people who know the standard definition. I'm not sure that's helpful. I also wouldn't think this we want this to be public.